horace_core/sqw/@sqw/private/cut_accumulate_data_.m

```diff
@@ -173,76 +173,76 @@
-        nbin_as_size = [nbin, 1];
-    else
-        nbin_as_size = nbin;
-    end
+% Get the given nbin array as a size
+
+% Note: Matlab silliness when one dimensional: MUST add an outer dimension
+% of unity. For 2D and higher, outer dimensions can always be assumed.
+% The problem with 1D is that e.g. zeros([5]) is not the same as
+% zeros([5,1]) whereas zeros([5,3]) is the same as zeros([5,3,1]).
+if isempty(nbin)
+    nbin_as_size = [1, 1];
+elseif length(nbin) == 1
+    nbin_as_size = [nbin, 1];
+else
+    nbin_as_size = nbin;
+end
 end
 
 
 function [s, e] = average_signal(s, e, npix)
-    % Convert summed signal & error into averages
-    s = s./npix;
-    e = e./(npix.^2);
-    no_pix = (npix == 0);  % true where no pixels contribute to given bin
-
-    % By convention, signal and error are zero if no pixels contribute to bin
-    s(no_pix) = 0;
-    e(no_pix) = 0;
+% Convert summed signal & error into averages
+s = s./npix;
+e = e./(npix.^2);
+no_pix = (npix == 0);  % true where no pixels contribute to given bin
+
+% By convention, signal and error are zero if no pixels contribute to bin
+s(no_pix) = 0;
+e(no_pix) = 0;
 end
 
 
 function pci = init_pix_combine_info(nfiles, nbins)
-    % Create a pix_combine_info object to manage temporary files of pixels
-    wk_dir = get(parallel_config, 'working_directory');
-    tmp_file_names = gen_array_of_tmp_file_paths(nfiles, wk_dir);
-    pci = pix_combine_info(tmp_file_names, nbins);
+% Create a pix_combine_info object to manage temporary files of pixels
+wk_dir = get(parallel_config, 'working_directory');
+tmp_file_names = gen_array_of_tmp_file_paths(nfiles, wk_dir);
+pci = pix_combine_info(tmp_file_names, nbins);
 end
 
 
 function paths = gen_array_of_tmp_file_paths(nfiles, base_dir)
-    % Generate a cell array of paths for temporary files to be written to
-    % Format of the file names follows:
-    %   horace_cut_<UUID>_<counter_with_padded_zeros>.tmp
-    if nfiles < 1
-        error('CUT:cut_accumulate_data_', ...
-              ['Cannot create temporary file paths for less than 1 file.' ...
-               '\nFound %i.'], nfiles);
-    end
-    prefix = 'horace_cut';
-    uuid = char(java.util.UUID.randomUUID());
-    counter_padding = floor(log10(nfiles)) + 1;
-    format_str = sprintf('%s_%s_%%0%ii.tmp', prefix, uuid, counter_padding);
-    paths = cell(1, nfiles);
-    for i = 1:nfiles
-        paths{i} = fullfile(base_dir, sprintf(format_str, i));
-    end
+% Generate a cell array of paths for temporary files to be written to
+% Format of the file names follows:
+%   horace_cut_<UUID>_<counter_with_padded_zeros>.tmp
+if nfiles < 1
+    error('CUT:cut_accumulate_data_', ...
+        ['Cannot create temporary file paths for less than 1 file.' ...
+        '\nFound %i.'], nfiles);
+end
+prefix = 'horace_cut';
+uuid = char(java.util.UUID.randomUUID());
+counter_padding = floor(log10(nfiles)) + 1;
+format_str = sprintf('%s_%s_%%0%ii.tmp', prefix, uuid, counter_padding);
```

Comment:
is this repeating "generation of filenames" code elsewhere?

Reply:
I have no idea __ I have not changed this code. May only formatted it according to Matlab formatting